jumanji/environments/logic/graph_coloring/generator.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+import chex
+import jax
+from jax import numpy as jnp
+
+
+class Generator(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def num_nodes(self) -> int:
+        """Number of nodes of the problem instances generated.
+        Returns:
+            `num_nodes` of the generated instances.
+        """
+
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey) -> chex.Array:
+        """Generate a problem instance.
+
+        Args:
+            key: random key.
+
+        Returns:
+            An `adj_matrix` representing a problem instance.
+        """
+
+
+class RandomGenerator(Generator):
+    """A generator for random graphs in the context of GraphColoring problems,
+    based on the Erdős-Rényi model (G(n, p)).
+
+    The adjacency matrix is generated such that the graph is undirected and loop-less.
+    The graph is generated with a specified number of nodes and percentage of connectivity,
+    which is used as a proxy for the edge probability in the Erdős-Rényi model.
+    """
+
+    def __init__(self, num_nodes: int, edge_probability: float):
+        """
+        Initialize the RandomGraphColoringGenerator.
```

Comment:
Can you double check that doc strings are done consistently?
```suggestion
        """Initialize the RandomGraphColoringGenerator.
```